File: interviewbit/math/trailing-zeroes-in-factorial.java
/*
    interviewbit
    track: "programming",
    section: "math",
    problem title: "Trailing Zeroes in Factorial"
*/

import java.lang.Math;
public class Solution {
    public int trailingZeroes(int A) {
        int maxPowFive = ((int)(Math.log(A)/Math.log(5)));
        int extraFives = 0;
        for(int i=2;i<=maxPowFive;i++){
            extraFives+=(A/Math.pow(5,i));
        }
        int fives = (int)(A/5);
        return fives+extraFives;
    }
}

File: interviewbit/math/trailing-zeroes-in-factorial.js
/*
	interviewbit
	track: "programming",
	section: "math",
	problem title: "Trailing Zeroes in Factorial"
*/

module.exports = { 
 //param A : integer
 //return an integer
	trailingZeroes : function(A){
	    var i=1,fivesCount=0;
        while(A>=Math.pow(5,i)){
            fivesCount+=Math.floor(A/(Math.pow(5,i)));
            i++;
        }
        return fivesCount;
	}
};